src/shogun/clustering/KMeansMiniBatch.cpp

```diff
@@ -121,13 +121,14 @@ SGVector<int32_t> CKMeansMiniBatch::mbchoose_rand(int32_t b, int32_t num)
 	return ret;
 }
 
-void CKMeansMiniBatch::init_mb_params()
-{
-	batch_size=100;
+	void CKMeansMiniBatch::init_mb_params()
+	{
+		batch_size = 100;
 
-	SG_ADD(
-		&batch_size, "batch_size", "batch size for mini-batch KMeans");
-}
+		SG_ADD(
+		    &batch_size, "batch_size", "batch size for mini-batch KMeans",
+		    ParameterProperties::HYPER);
```

Comment:
@karlnapf  hyper or setting?

Reply:
Both?

Reply:
OK! there are quite a few batch_size parameters in different algos, so let's say always hyper and setting for those?

Reply:
well if we write some automatic hyperparameter tuning algorithm it would probably be a waste of resources to optimise batch size no? I mean it has an impact but if you choose a reasonable value it shouldn't be that important in prediction performance no? That's why i was thinking just setting

Reply:
but the batch size needs to be tuned indeed for certain methods (batch size often relates to regularization so it can have a big impact)..... and of course we will have methods to exclude parameters from optimizing..... After thinking about it I think it should be both

Reply:
ok!